R/charlson_comorbidity_index.R

```diff
@@ -124,67 +123,103 @@ diagnoses_at_admission <- function(ipdiag, erdiag) {
 
 
 #' @title
-#' Compute Charlson comorbidity index (CCI) score
+#' Compute Comorbidity Index
 #'
 #' @description
-#' Based on the methodology from [Quan et al. 2011](https://pubmed.ncbi.nlm.nih.gov/21330339/)
-#' using ICD-10 codes as the mapping algorithm,
-#' implemented in the R [comorbidity](https://ellessenne.github.io/comorbidity/index.html) package
-#' by [Gasparini, 2018](https://doi.org/10.21105/joss.00648).
-#'
-#' Can compute either CCI score at admission based on
-#' [HSMR Methodology](https://www.cihi.ca/sites/default/files/document/hospital-standardized-mortality-ratio-meth-notes-en.pdf)
-#' or for the entire patient encounter. The default is to compute comorbidity at admission.
+#' This is a generic function that is wrapped by [Rgemini::charlson_comorbidity_index()]
+#' and [Rgemini::elixhauser_comorbidity_index()] and is not meant to be used directly.
+#' The function is an interface to [comorbidity::comorbidity()] for GEMINI data,
+#' providing options for outputting raw comorbidities or comorbidities at admission.
 #'
 #' @inheritParams diagnoses_at_admission
+#' @inheritParams comorbidity::comorbidity
+#' @inheritParams comorbidity::score
 #'
 #' @param raw_comorbidities (`logical`)\cr
-#' Whether to output a `data.table` of raw comorbidities as opposed to pre-computed charlson scores.
+#' Whether to output a `data.table` of raw comorbidities as opposed to pre-computed scores.
 #'
 #' @param at_admission (`logical`)\cr
-#' Whether to calculate the charlson score for all comorbidities or for only pre-admit comorbidities.
+#' Whether to calculate the score for all comorbidities or for only pre-admit comorbidities.
 #'
 #' @return (`data.table`)\cr
-#' By default, for each encounter, outputs the Charlson comorbidity index. If `raw_comorbidities` is `TRUE`,
+#' By default, for each encounter, outputs the comorbidity score. If `raw_comorbidities` is `TRUE`,
 #' outputs a wide `data.table` with a column for each comorbidity for each encounter.
 #'
-#' @note
-#' A hierarchy of comorbidities is used when calculating the Charlson score, but not when outputting `raw_comorbidites`.
+#' @details
+#' A hierarchy of comorbidities is used when calculating the cormorbidity score, but not when outputting `raw_comorbidites`.
 #' This affects comorbidities present in a patient with different degrees of severity. See documentation for
 #' [comorbidity::comorbidity()] for details.
 #'
 #' @importFrom comorbidity comorbidity score
 #' @export
 #'
-charlson_comorbidity_index <- function(ipdiag, erdiag, at_admission = TRUE, raw_comorbidities = FALSE) {
+comorbidity_index <- function(ipdiag, erdiag, map, weights, at_admission = TRUE, raw_comorbidities = FALSE) {
   diagnoses <- if (at_admission) {
     diagnoses_at_admission(ipdiag, erdiag)
   } else {
     rbind(
       erdiag[, .(genc_id, "diagnosis_code" = er_diagnosis_code, "diagnosis_type" = er_diagnosis_type)],
```

Comment:
When `at_admission = TRUE`, the `diagnoses_at_admission` coerce the input `ipdiag` and `erdiag` to `data.table`s but when `at_admission = FALSE` this step is missing, and as a result when `at_admission = FALSE` and when either input is `data.frame` the function stops.